src/library/VUMeter.js

```diff
@@ -0,0 +1,119 @@
+import FIFO from './FIFO.js';
+
+/**
+ * This VUMeter class implements a volume level meter UI. To
+ * initialize it, developers need to provide the ID of the canvas
+ * element where the VUMeter will be rendered, the minimum decibel
+ * value for the VUMeter, an AnalyserNode, fftSize of the
+ * analyserNode, and fifoSize to control the VMMeter update speed. The
+ * minimum decibel value defines the range of the VUMeter that will be
+ * considered silent. When the VUMeter is created, it requires an
+ * AnalyserNode for real-time audio input data. This class requires a
+ * canvas element and FIFO class. The FIFO class is used to determine
+ * how fast the VUMeter gets updated. The Waveform bar size is
+ * determined by the minimum number in the FIFO class.
+ * @class
+ */
+class VUMeter {
+
+  /**
+   * @constructor Initializes the VUMeter with the specified canvas
+   * ID, minimum decibel value, analyserNode, fftSize, and fifoSize.
+   * @param {String} canvasId A canvas element ID where the
+   * visualization is drawn to.
+   * @param {Number} minDecibel The minimum lower bound decibel of the
+   * visualization. If the measured volume is lower than this value,
+   * it is considered silent.
+   * @param {AnalyserNode} analyserNode The AnalyserNode used for the
+   * visualization.
+   * @param {Number} fftSize The window size in samples used for
+   * performing a Fast Fourier Transform (FFT) to obtain frequency
+   * domain data in the AnalyserNode.
+   * @param {Number} fifoSize The size of the FIFO queue which
+   * determines how fast the VUMeter gets updated.
+   */
+  constructor(canvasId, minDecibel, analyserNode, fftSize, fifoSize) {
+    /** @private @const {!canvas} Selected VUMeter canvas element */
+    this.canvas_ = document.querySelector(canvasId);
+    /** @private @const {!canvasContext} Canvas context */
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    /** @private @const {!width} Selected VUMeter canvas width */
+    this.width_ = this.canvas_.width;
+    /** @private @const {!height} Selected VUMeter canvas height */
+    this.height_ = this.canvas_.height;
+    /** @private @const {!minDisplayDecibel} Minimum decibel level for
+     * the VU meter. Any decibel lower than this value is considered
+     * silent and converted to a positive value for calculation and
+     * graph convenience. */
+    this.minDisplayDecibel_ = Math.abs(minDecibel);
+    /** @private @const {!fifo} First-in-first-out queue for
```

Comment:
Here you do not need to explain what FIFO is. Just mention FIFO in the overview above. Also we are not using this specifically for "controlling updating speed" and it's a byproduct of average data collected in the FIFO.

Perhaps:

`An instance of FIFO class for caching RMS values and extracting a minimum value over time.`